### PR TITLE
Setup redux

### PR DESCRIPTION
**Pull request by Yetidancer**
# Description :: User Story

As a User;
On the homepage;
I will see the following:
-an input for "name" which will be the user's chat name, which throws an error if left blank
-an input for "group code", which throws an error if left blank or incorrectly filled out
-a button for "join room" which will be disabled until form filled that will take me to a chat with a random group member
-a button to "create group" which, when clicked, will take me to a new group form
-EXT. profanity filter for name submission

## Type of change

- [x] New Feature
- [ ] Bug Fix
- [x] Refactor/setup
- [ ] Breaking Change

## Notes

This pull requests sets up redux and adds feature to display rooms in dropdown that user has already entered.

### CODE REVIEW

**Comment by KmBarnett** [comment]
https://frontend.turing.io/lessons/module-3/testing-redux.html
https://frontend.turing.io/lessons/module-3/redux-i.html

Resources for further dive into redux

File: src/Components/JoinForm/JoinForm.js [hunk 1]
```diff
@@ -66,4 +70,8 @@ const mapStateToProps = state => ({
   codes: state.codes,
 })
 
-export default connect(mapStateToProps, null)(JoinForm);
+const mapDispatchToProps = dispatch => ({
+  addNewCode: code => dispatch(addNewCode(code))
+})
+
+export default connect(mapStateToProps, mapDispatchToProps)(JoinForm);
```
Comment (Yetidancer) [inline src/Components/JoinForm/JoinForm.js:77]: what is this line doing?

File: src/Components/JoinForm/JoinForm.js [hunk 2]
```diff
@@ -2,13 +2,17 @@ import React, { useState, useEffect } from 'react';
...
-function JoinForm({ codes }) {
+function JoinForm({ codes, addNewCode }) {
   const [ nickname, setNickname ] = useState('')
   const [ roomCode, setRoomCode ] = useState('')
 
   const saveMyCodes = () => {
-    window.localStorage.setItem('codes', JSON.stringify( codes ))
+    addNewCode(roomCode)
+    if ( !codes.includes(roomCode)) {
+      window.localStorage.setItem('codes', JSON.stringify( [roomCode, ...codes] ))
```
Comment (Yetidancer) [inline src/Components/JoinForm/JoinForm.js:14]: this is what allows us to add entered codes into the local storage?

File: src/Components/JoinForm/JoinForm.js (hunk 1 above)
Comment (KmBarnett) [reply src/Components/JoinForm/JoinForm.js:77]: This syntax and line connects the component to the redux store defined in the provider in src index.js

File: src/Components/App/App.js [hunk 3]
```diff
@@ -1,11 +1,26 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import './App.css';
 import JoinForm from '../JoinForm/JoinForm.js';
 import RoomForm from '../RoomForm/RoomForm.js';
 import { Route } from 'react-router-dom';
-import {ReactComponent as ChatLogo} from './assets/message.svg'
+import {ReactComponent as ChatLogo} from './assets/message.svg';
+import { getCircleCode } from './../../Actions/index.js';
```
Comment (Yetidancer) [inline src/Components/App/App.js]: is there better syntax for the path? looks kinda gross no offense

File: src/Components/JoinForm/JoinForm.js (hunk 2 above)
Comment (KmBarnett) [reply src/Components/JoinForm/JoinForm.js:14]: Yeah, this checks that it's a new code that local storage doesn't have and then overwrites local with the updated list.

File: src/index.js [hunk 4]
```diff
@@ -9,12 +9,16 @@ import { createStore } from 'redux';
 import { composeWithDevTools } from 'redux-devtools-extension';
 import rootReducer from './Reducers';
 
+const store = createStore(rootReducer, composeWithDevTools())
+
+
+
 ReactDOM.render(
-  <Router>
-    <React.StrictMode>
```
Comment (Yetidancer) [inline src/index.js:10]: after all that... no strict mode?

File: src/Components/App/App.js (hunk 3 above)
Comment (KmBarnett) [reply src/Components/App/App.js]: Not that I know of.

File: src/index.js (hunk 4 above)
Comment (KmBarnett) [reply src/index.js:10]: https://reactjs.org/docs/strict-mode.html
not sure why it needed not to much experience with it.

**Pull request merged**